File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    hash_dict = {}
    sum_hash = 0
    for part in participant:
        hash_dict[hash(part)] = part
        sum_hash += hash(part)

    for comp in completion:
        sum_hash -= hash(comp)

    return hash_dict[sum_hash]

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<Integer, String> dict = new HashMap<>();
        int sumHash = 0;
        
        for (String part : participant) {
            dict.put(part.hashCode(), part);
            sumHash += part.hashCode();
        }
        for (String comp : completion) {
            sumHash -= comp.hashCode();
        }
        return dict.get(sumHash);
        
    }
}